# Introduction

## Problem

Some Time when a person he / she does not know that place very well so it is very hard for him to find an hospital at the time of emergency some time person may get know the name of the hospital from another person but he does not know about the quality of that particular hospital so in this project we make hospital recomendation system where user will put the detail of the level of quality of hospital and as per his/her quality level we recomended hospital to him .

Now by using ***Foursquare location*** this API is used to show the location of those recomended hospital in the world map 





## Background

Finding a hospital is different than to find a good hospital some time we are not able to find best hospital at the time of emergency in google map because in google map we can not provide some parameter such as Meets criteria for promoting interoperability of EHRs ,Mortality national comparison footnote,Safety of care national comparison footnote,Readmission national comparisonn etc so for that we created an project where we can use these feature to recomend a new hospital for our user 


# Data

Here we are using the data set that contain the folloing things 
<li>
Faculty Id 
<li>
Facility Name
    <li>
Address
 <li>
City
  <li>
State
   <li>
ZIP Code
       <li>
County Name
           <li>
Phone Number
               <li>
Hospital Type
                   <li>
Hospital Ownership
                       <li>
Emergency Services
                           <li>
Meets criteria for promoting interoperability of EHRs
                               <li>
Hospital overall ratingn
                                   <li>
Hospital overall rating footnote
                                       <li>
Mortality national comparisonn
                                           <li>
Mortality national comparison footnote
                                               <li>
Safety of care national comparisonn
                                                   <li>
Safety of care national comparison footnote
                                                       <li>
Readmission national comparisonn
                                                           <li>
Readmission national comparison footnote
                                                               <li>
Patient experience national comparisonn
                                                                   <li>
Patient experience national comparison footnote
                                                                       <li>
Effectiveness of care national comparison footnoten
                                                                           <li>
Effectiveness of care national comparison footnoten
                                                                               <li>
Timeliness of care national comparisonn
                                                                                   <li>
Timeliness of care national comparison footnote
                                                                                       <li>
Efficient use of medical imaging national comparison footnote
                                                                                           <li>
Location
<li>
Among The above factor there some factor which decide the quality of the hospital among above features the feature which determine the quality of hospital is given below 

Emergency Services
                           <li>
Meets criteria for promoting interoperability of EHRs
                               <li>
Hospital overall ratingn
                                   <li>
Hospital overall rating footnote
                                       <li>
Mortality national comparisonn
                                           <li>
Mortality national comparison footnote
                                               <li>
Safety of care national comparisonn
                                                   <li>
Safety of care national comparison footnote
                                                       <li>
Readmission national comparisonn
                                                           <li>
Readmission national comparison footnote
                                                               <li>
Patient experience national comparisonn
                                                                   <li>
Patient experience national comparison footnote
                                                                       <li>
Effectiveness of care national comparison footnoten
                                                                           <li>
Effectiveness of care national comparison footnoten
                                                                               <li>
Timeliness of care national comparisonn
                                                                                   <li>
Timeliness of care national comparison footnote
                                                                                       <li>
Efficient use of medical imaging national comparison footnote
                                                                                           <li>
Location
                                                                                               
User have to provides thes data and among these data our recomender system will recomende the hospital suppose hospital recomendaded by our model is of Hospital Type Psychiatric then all the hospital of these type will be shown in the map by using ***Foursquare location*** t 
 


# import libary

In [6]:
import pandas as pd
import os
import numpy as np 
from sklearn.tree import DecisionTreeClassifier

#### Download the data set from my ibm storage which i store for this project 

In [7]:

# @hidden_cell
# The following code contains the credentials for a file in your IBM Cloud Object Storage.
# You might want to remove those credentials before you share your notebook.
credentials_1 = {
    'IAM_SERVICE_ID': 'iam-ServiceId-b21f1762-43ac-4c08-b713-87cc313601ba',
    'IBM_API_KEY_ID': 'RxZH6HW3U6L9FJUUv4Z53BJeVVYdZ4yvYvPrX7tpEG0P',
    'ENDPOINT': 'https://s3-api.us-geo.objectstorage.service.networklayer.com',
    'IBM_AUTH_ENDPOINT': 'https://iam.cloud.ibm.com/oidc/token',
    'BUCKET': 'hospitalselection-donotdelete-pr-qrw5w2z8u3gbmq',
    'FILE': 'Hospital_General_Information.csv'
}


In [8]:

import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
client_8019663306054913a106f7ca1a38c7b3 = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='RxZH6HW3U6L9FJUUv4Z53BJeVVYdZ4yvYvPrX7tpEG0P',
    ibm_auth_endpoint="https://iam.cloud.ibm.com/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3-api.us-geo.objectstorage.service.networklayer.com')

body = client_8019663306054913a106f7ca1a38c7b3.get_object(Bucket='hospitalselection-donotdelete-pr-qrw5w2z8u3gbmq',Key='Hospital_General_Information.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

df_data_1 = pd.read_csv(body)
df_data_1.head()


Facility ID                            Facility Name  \
0       44022                 CONWAY BEHAVIORAL HEALTH   
1       54154               SAN JOSE BEHAVIORAL HEALTH   
2      121300         KAUAI VETERANS MEMORIAL HOSPITAL   
3      141328  HARDIN COUNTY GENERAL HOSPITAL & CLINIC   
4      154060                         PARK CENTER, INC   

                        Address        City State  ZIP Code  County Name  \
0             2255 STURGIS ROAD      CONWAY    AR     72034     FAULKNER   
1  455 SILICON VALLEY BOULEVARD    SAN JOSE    CA     95138  SANTA CLARA   
2      4643 WAIMEA CANYON DRIVE      WAIMEA    HI     96796        KAUAI   
3                  FERRELL ROAD   ROSICLARE    IL     62982       HARDIN   
4             1909 CAREW STREET  FORT WAYNE    IN     46805        ALLEN   

   Phone Number              Hospital Type              Hospital Ownership  \
0  501 205-0011                Psychiatric                     Proprietary   
1  669 234-5959                Psychiatric                     Proprietary   
2  808 338-9431  Critical Access Hospitals              Government - State   
3  618 285-6634  Critical Access Hospitals  Voluntary non-profit - Private   
4  260 481-2700                Psychiatric  Voluntary non-profit - Private   

   ... Readmission national comparisonn  \
0  ...                    Not Available   
1  ...                    Not Available   
2  ...                    Not Available   
3  ...       Below the national average   
4  ...                    Not Available   

  Readmission national comparison footnote  \
0                                     19.0   
1                                     19.0   
2                                      5.0   
3                                      NaN   
4                                     19.0   

  Patient experience national comparisonn  \
0                           Not Available   
1                           Not Available   
2                           Not Available   
3                           Not Available   
4                           Not Available   

   Patient experience national comparison footnote  \
0                                             19.0   
1                                             19.0   
2                                             16.0   
3                                             16.0   
4                                             19.0   

  Effectiveness of care national comparison footnoten  \
0                                      Not Available    
1                                      Not Available    
2                       Same as the national average    
3                                      Not Available    
4                                      Not Available    

   Effectiveness of care national comparison footnoten.1  \
0                                               19.0       
1                                               19.0       
2                                                NaN       
3                                               16.0       
4                                               19.0       

  Timeliness of care national comparisonn  \
0                           Not Available   
1                           Not Available   
2            Same as the national average   
3                           Not Available   
4                           Not Available   

   Timeliness of care national comparison footnote  \
0                                             19.0   
1                                             19.0   
2                                              NaN   
3                                              5.0   
4                                             19.0   

  Efficient use of medical imaging national comparison footnote  \
0                                               19.0              
1                                               19.0              
2                                               16.0              
3                          

#### Data preprocing that is cleaning of the data drop those table which are not required but before droping it stored it in your data set  in ibm cloud

In [10]:
df_data_1.drop('Mortality national comparisonn',axis=1,inplace=True)
df_data_1.drop('Safety of care national comparisonn',axis=1,inplace=True)
df_data_1.drop('Readmission national comparisonn',axis=1,inplace=True)
df_data_1.drop('Patient experience national comparisonn',axis=1,inplace=True)
df_data_1.drop('Effectiveness of care national comparison footnoten',axis=1,inplace=True)


In [12]:
#df_data_1.head()

In [13]:
df_data_1.drop('Timeliness of care national comparisonn',axis=1,inplace=True)

In [15]:
df_data_1.head()

In [16]:
#df_data_1.drop('Hospital overall ratingn',axis=1,inplace=True)

In [18]:
df_data_1.head()

In [19]:
df_data_1.head()

Facility ID                            Facility Name  \
0       44022                 CONWAY BEHAVIORAL HEALTH   
1       54154               SAN JOSE BEHAVIORAL HEALTH   
2      121300         KAUAI VETERANS MEMORIAL HOSPITAL   
3      141328  HARDIN COUNTY GENERAL HOSPITAL & CLINIC   
4      154060                         PARK CENTER, INC   

                        Address        City State  ZIP Code  County Name  \
0             2255 STURGIS ROAD      CONWAY    AR     72034     FAULKNER   
1  455 SILICON VALLEY BOULEVARD    SAN JOSE    CA     95138  SANTA CLARA   
2      4643 WAIMEA CANYON DRIVE      WAIMEA    HI     96796        KAUAI   
3                  FERRELL ROAD   ROSICLARE    IL     62982       HARDIN   
4             1909 CAREW STREET  FORT WAYNE    IN     46805        ALLEN   

   Phone Number              Hospital Type              Hospital Ownership  \
0  501 205-0011                Psychiatric                     Proprietary   
1  669 234-5959                Psychiatric                     Proprietary   
2  808 338-9431  Critical Access Hospitals              Government - State   
3  618 285-6634  Critical Access Hospitals  Voluntary non-profit - Private   
4  260 481-2700                Psychiatric  Voluntary non-profit - Private   

   ... Meets criteria for promoting interoperability of EHRs  \
0  ...                                                NaN      
1  ...                                                NaN      
2  ...                                                  Y      
3  ...                                                  Y      
4  ...                                                NaN      

  Hospital overall rating footnote  Mortality national comparison footnote  \
0                             19.0                                    19.0   
1                             19.0                                    19.0   
2                             16.0                                    16.0   
3                             16.0                                     5.0   
4                             19.0                                    19.0   

   Safety of care national comparison footnote  \
0                                         19.0   
1                                         19.0   
2                                         16.0   
3                                         16.0   
4                                         19.0   

   Readmission national comparison footnote  \
0                                      19.0   
1                                      19.0   
2                                       5.0   
3                                       NaN   
4                                      19.0   

   Patient experience national comparison footnote  \
0                                             19.0   
1                                             19.0   
2                                             16.0   
3                                             16.0   
4                                             19.0   

   Effectiveness of care national comparison footnoten.1  \
0                                               19.0       
1                                               19.0       
2                                                NaN       
3                                               16.0       
4                                               19.0       

   Timeliness of care national comparison footnote  \
0                                             19.0   
1                                             19.0   
2                                              NaN   
3                                              5.0   
4                                             19.0   

   Efficient use of medical imaging national comparison footnote  \
0                                               19.0               
1                                               19.0               
2                                               16.0               
3     

#### drop nan value

In [20]:
df_data_1.dropna(subset=["Hospital Type"], axis=0, inplace=True)
df_data_1.dropna(subset=["Emergency Services"], axis=0, inplace=True)
df_data_1.dropna(subset=["Meets criteria for promoting interoperability of EHRs"], axis=0, inplace=True)
df_data_1.dropna(subset=["Hospital overall rating footnote"], axis=0, inplace=True)
df_data_1.dropna(subset=["Mortality national comparison footnote"], axis=0, inplace=True)
df_data_1.dropna(subset=["Safety of care national comparison footnote"], axis=0, inplace=True)
df_data_1.dropna(subset=["Readmission national comparison footnote"], axis=0, inplace=True)
df_data_1.dropna(subset=["Patient experience national comparison footnote"], axis=0, inplace=True)
df_data_1.dropna(subset=["Effectiveness of care national comparison footnoten.1"], axis=0, inplace=True)
df_data_1.dropna(subset=["Timeliness of care national comparison footnote"], axis=0, inplace=True)
df_data_1.dropna(subset=["Efficient use of medical imaging national comparison footnote"], axis=0, inplace=True)
df_data_1.dropna(subset=["Location"], axis=0, inplace=True)

In [21]:
df_data_1.reset_index(drop=True, inplace=True)

In [22]:
df_data_1.head()

Facility ID                      Facility Name                 Address  \
0      271328              MARIAS MEDICAL CENTER            640 PARK AVE   
1       51335     BEAR VALLEY COMMUNITY HOSPITAL        41870 GARSTIN DR   
2       61343            KEEFE MEMORIAL HOSPITAL          602 N 6TH W ST   
3      431307  SANFORD CLEAR LAKE MEDICAL CENTER  701 THIRD AVENUE SOUTH   
4      381310  PIONEER MEMORIAL HOSPITAL HEPPNER     564 E PIONEER DRIVE   

             City State  ZIP Code     County Name  Phone Number  \
0          SHELBY    MT     59474           TOOLE  406 434-3200   
1   BIG BEAR LAKE    CA     92315  SAN BERNARDINO  909 866-6501   
2  CHEYENNE WELLS    CO     80810        CHEYENNE  719 767-5661   
3      CLEAR LAKE    SD     57226           DEUEL  605 874-2141   
4         HEPPNER    OR     97836          MORROW  541 676-9133   

               Hospital Type                           Hospital Ownership  \
0  Critical Access Hospitals                           Government - Local   
1  Critical Access Hospitals  Government - Hospital District or Authority   
2  Critical Access Hospitals  Government - Hospital District or Authority   
3  Critical Access Hospitals               Voluntary non-profit - Private   
4  Critical Access Hospitals  Government - Hospital District or Authority   

   ... Meets criteria for promoting interoperability of EHRs  \
0  ...                                                  Y      
1  ...                                                  Y      
2  ...                                                  Y      
3  ...                                                  Y      
4  ...                                                  Y      

  Hospital overall rating footnote  Mortality national comparison footnote  \
0                             16.0                                    16.0   
1                             16.0                                    16.0   
2                             16.0                                    16.0   
3                             16.0                                    16.0   
4                             16.0                                    16.0   

   Safety of care national comparison footnote  \
0                                         16.0   
1                                         16.0   
2                                         16.0   
3                                         16.0   
4                                         16.0   

   Readmission national comparison footnote  \
0                                       5.0   
1                                       5.0   
2                                      16.0   
3                                       5.0   
4                                       5.0   

   Patient experience national comparison footnote  \
0                                             16.0   
1                                             16.0   
2                                             16.0   
3                                             16.0   
4                                             16.0   

   Effectiveness of care national comparison footnoten.1  \
0                                                5.0       
1                                                5.0       
2                                               16.0       
3                                               16.0       
4                                               16.0       

   Timeliness of care national comparison footnote  \
0                                              5.0   
1                                              5.0   
2                                              5.0   
3                                             16.0   
4                                              5.0   

   Efficient use of medical imaging national comparison footnote  \
0                                               16.0               
1                                               16.0               
2                         

#### convert categorical value into 0 and 1 here for yes i used 0 and for no i used 1

In [23]:
dummy=pd.get_dummies(df_data_1['Emergency Services'])
dummy

No  Yes
0     0    1
1     1    0
2     0    1
3     0    1
4     0    1
..   ..  ...
112   0    1
113   1    0
114   0    1
115   0    1
116   0    1

[117 rows x 2 columns]

In [24]:
df_data_2=pd.concat((df_data_1,dummy),axis=1)

In [25]:
df_data_2=df_data_2.drop(['Emergency Services'],axis=1)

In [26]:
df_data_2=df_data_2.drop(['Yes'],axis=1)

In [27]:
df_data_2=df_data_2.rename(columns={"No":"Emergency Services"})

In [28]:
df_data_2.drop('Meets criteria for promoting interoperability of EHRs',axis=1,inplace=True)

In [29]:
df_data_2

Facility ID                      Facility Name  \
0        271328              MARIAS MEDICAL CENTER   
1         51335     BEAR VALLEY COMMUNITY HOSPITAL   
2         61343            KEEFE MEMORIAL HOSPITAL   
3        431307  SANFORD CLEAR LAKE MEDICAL CENTER   
4        381310  PIONEER MEMORIAL HOSPITAL HEPPNER   
..          ...                                ...   
112      451390           DIMMIT REGIONAL HOSPITAL   
113      441300      SAINT THOMAS HICKMAN HOSPITAL   
114      451351      LYNN COUNTY HOSPITAL DISTRICT   
115      181332            FLEMING COUNTY HOSPITAL   
116      670102              AD HOSPITAL EAST, LLC   

                           Address             City State  ZIP Code  \
0                     640 PARK AVE           SHELBY    MT     59474   
1                 41870 GARSTIN DR    BIG BEAR LAKE    CA     92315   
2                   602 N 6TH W ST   CHEYENNE WELLS    CO     80810   
3           701 THIRD AVENUE SOUTH       CLEAR LAKE    SD     57226   
4              564 E PIONEER DRIVE          HEPPNER    OR     97836   
..                             ...              ...   ...       ...   
112             704 HOSPITAL DRIVE  CARRIZO SPRINGS    TX     78834   
113           135 EAST SWAN STREET      CENTERVILLE    TN     37033   
114           2600 LOCKWOOD STREET           TAHOKA    TX     79373   
115            55 FOUNDATION DRIVE     FLEMINGSBURG    KY     41041   
116  12950 EAST FREEWAY, SUITE 100          HOUSTON    TX     77015   

        County Name  Phone Number              Hospital Type  \
0             TOOLE  406 434-3200  Critical Access Hospitals   
1    SAN BERNARDINO  909 866-6501  Critical Access Hospitals   
2          CHEYENNE  719 767-5661  Critical Access Hospitals   
3             DEUEL  605 874-2141  Critical Access Hospitals   
4            MORROW  541 676-9133  Critical Access Hospitals   
..              ...           ...                        ...   
112          DIMMIT  830 876-2424  Critical Access Hospitals   
113         HICKMAN  931 729-4271  Critical Access Hospitals   
114            LYNN  806 998-4533  Critical Access Hospitals   
115         FLEMING  606 849-2351  Critical Access Hospitals   
116          HARRIS  713 330-3897       Acute Care Hospitals   

                              Hospital Ownership  \
0                             Government - Local   
1    Government - Hospital District or Authority   
2    Government - Hospital District or Authority   
3                 Voluntary non-profit - Private   
4    Government - Hospital District or Authority   
..                                           ...   
112                 Voluntary non-profit - Other   
113               Voluntary non-profit - Private   
114  Government - Hospital District or Authority   
115                                  Proprietary   
116                                  Proprietary   

     Hospital overall rating footnote  Mortality national comparison footnote  \
0                                16.0                                    16.0   
1                                16.0                                    16.0   
2                                16.0                                    16.0   
3                                16.0                                    16.0   
4                                16.0                                    16.0   
..                                ...                                     ...   
112                              16.0                                    16.0   
113                              16.0                                    16.0   
114                              16.0                                    16.0   
115                               5.0                                     5.0   
116                              16.0                                    16.0   

     Safety of care national comparison footnote  \
0                                           16.0   
1                              

#### make the hospital type coloumn in the the last which we have to predict accoring to the user quality level 

In [30]:
title=list(df_data_2.columns)
title

['Facility ID',
 'Facility Name',
 'Address',
 'City',
 'State',
 'ZIP Code',
 'County Name',
 'Phone Number',
 'Hospital Type',
 'Hospital Ownership',
 'Hospital overall rating footnote',
 'Mortality national comparison footnote',
 'Safety of care national comparison footnote',
 'Readmission national comparison footnote',
 'Patient experience national comparison footnote',
 'Effectiveness of care national comparison footnoten.1',
 'Timeliness of care national comparison footnote',
 'Efficient use of medical imaging national comparison footnote',
 'Location',
 'Emergency Services']

In [31]:
title[0],title[10]=title[10],title[0]

In [32]:
df_data_2=df_data_2[title]

In [33]:
df_data_2

Hospital overall rating footnote                      Facility Name  \
0                                16.0              MARIAS MEDICAL CENTER   
1                                16.0     BEAR VALLEY COMMUNITY HOSPITAL   
2                                16.0            KEEFE MEMORIAL HOSPITAL   
3                                16.0  SANFORD CLEAR LAKE MEDICAL CENTER   
4                                16.0  PIONEER MEMORIAL HOSPITAL HEPPNER   
..                                ...                                ...   
112                              16.0           DIMMIT REGIONAL HOSPITAL   
113                              16.0      SAINT THOMAS HICKMAN HOSPITAL   
114                              16.0      LYNN COUNTY HOSPITAL DISTRICT   
115                               5.0            FLEMING COUNTY HOSPITAL   
116                              16.0              AD HOSPITAL EAST, LLC   

                           Address             City State  ZIP Code  \
0                     640 PARK AVE           SHELBY    MT     59474   
1                 41870 GARSTIN DR    BIG BEAR LAKE    CA     92315   
2                   602 N 6TH W ST   CHEYENNE WELLS    CO     80810   
3           701 THIRD AVENUE SOUTH       CLEAR LAKE    SD     57226   
4              564 E PIONEER DRIVE          HEPPNER    OR     97836   
..                             ...              ...   ...       ...   
112             704 HOSPITAL DRIVE  CARRIZO SPRINGS    TX     78834   
113           135 EAST SWAN STREET      CENTERVILLE    TN     37033   
114           2600 LOCKWOOD STREET           TAHOKA    TX     79373   
115            55 FOUNDATION DRIVE     FLEMINGSBURG    KY     41041   
116  12950 EAST FREEWAY, SUITE 100          HOUSTON    TX     77015   

        County Name  Phone Number              Hospital Type  \
0             TOOLE  406 434-3200  Critical Access Hospitals   
1    SAN BERNARDINO  909 866-6501  Critical Access Hospitals   
2          CHEYENNE  719 767-5661  Critical Access Hospitals   
3             DEUEL  605 874-2141  Critical Access Hospitals   
4            MORROW  541 676-9133  Critical Access Hospitals   
..              ...           ...                        ...   
112          DIMMIT  830 876-2424  Critical Access Hospitals   
113         HICKMAN  931 729-4271  Critical Access Hospitals   
114            LYNN  806 998-4533  Critical Access Hospitals   
115         FLEMING  606 849-2351  Critical Access Hospitals   
116          HARRIS  713 330-3897       Acute Care Hospitals   

                              Hospital Ownership Facility ID  \
0                             Government - Local      271328   
1    Government - Hospital District or Authority       51335   
2    Government - Hospital District or Authority       61343   
3                 Voluntary non-profit - Private      431307   
4    Government - Hospital District or Authority      381310   
..                                           ...         ...   
112                 Voluntary non-profit - Other      451390   
113               Voluntary non-profit - Private      441300   
114  Government - Hospital District or Authority      451351   
115                                  Proprietary      181332   
116                                  Proprietary      670102   

     Mortality national comparison footnote  \
0                                      16.0   
1                                      16.0   
2                                      16.0   
3                                      16.0   
4                                      16.0   
..                                      ...   
112                                    16.0   
113                                    16.0   
114                                    16.0   
115                                     5.0   
116                                    16.0   

     Safety of care national comparison footnote  \
0                                           16.0   
1                               

#### Enter the features set based on which my project recomend the type of hospital 

In [34]:
X = df_data_2[['Emergency Services','Hospital overall rating footnote','Mortality national comparison footnote','Safety of care national comparison footnote','Readmission national comparison footnote','Patient experience national comparison footnote','Effectiveness of care national comparison footnoten.1','Timeliness of care national comparison footnote','Efficient use of medical imaging national comparison footnote']].values

In [35]:
X

array([[ 0., 16., 16., ...,  5.,  5., 16.],
       [ 1., 16., 16., ...,  5.,  5., 16.],
       [ 0., 16., 16., ..., 16.,  5., 16.],
       ...,
       [ 0., 16., 16., ..., 16., 16.,  5.],
       [ 0.,  5.,  5., ...,  5.,  5.,  5.],
       [ 0., 16., 16., ..., 16., 16., 16.]])

#### my target varivable which i want to predict 

In [36]:
y = df_data_2["Hospital Type"]

In [37]:
from sklearn.model_selection import train_test_split

#### Divide data set into two part test and train to check our model acuracy

In [38]:
X_trainset, X_testset, y_trainset, y_testset = train_test_split(X, y, test_size=0.3, random_state=3)

In [39]:
drugTree = DecisionTreeClassifier(criterion="entropy", max_depth = 4)
drugTree

DecisionTreeClassifier(criterion='entropy', max_depth=4)

In [40]:
drugTree.fit(X_trainset,y_trainset)

DecisionTreeClassifier(criterion='entropy', max_depth=4)

In [41]:
predTree = drugTree.predict(X_testset)
predTree

array(['Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Acute Care Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Acute Care Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Acute Care Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Critical Access Hospitals',
       'Critical Access Hospitals', 'Cri

In [42]:
from sklearn import metrics
import matplotlib.pyplot as plt
print("DecisionTrees's Accuracy: ", metrics.accuracy_score(y_testset, predTree))

DecisionTrees's Accuracy:  0.9444444444444444


we get an acuaracy of 0.944 which is an good so we can proceed with our model 

##### user put the quality level that tell what type of model needed to use 

In [43]:
XTEST=[[0,2,4,7,5,15,4,7,8]]

In [44]:
P=drugTree.predict(XTEST)
P[0]

'Critical Access Hospitals'

now explore the adreeses of those hospital which have hospital type Critical Access Hospitals by exploring these addreses we came to know what are the best place to take food to take rest 

#### before exploring the hospital let add those data which i have drop for prediction our hospital type 

In [45]:
df_data_2

Hospital overall rating footnote                      Facility Name  \
0                                16.0              MARIAS MEDICAL CENTER   
1                                16.0     BEAR VALLEY COMMUNITY HOSPITAL   
2                                16.0            KEEFE MEMORIAL HOSPITAL   
3                                16.0  SANFORD CLEAR LAKE MEDICAL CENTER   
4                                16.0  PIONEER MEMORIAL HOSPITAL HEPPNER   
..                                ...                                ...   
112                              16.0           DIMMIT REGIONAL HOSPITAL   
113                              16.0      SAINT THOMAS HICKMAN HOSPITAL   
114                              16.0      LYNN COUNTY HOSPITAL DISTRICT   
115                               5.0            FLEMING COUNTY HOSPITAL   
116                              16.0              AD HOSPITAL EAST, LLC   

                           Address             City State  ZIP Code  \
0                     640 PARK AVE           SHELBY    MT     59474   
1                 41870 GARSTIN DR    BIG BEAR LAKE    CA     92315   
2                   602 N 6TH W ST   CHEYENNE WELLS    CO     80810   
3           701 THIRD AVENUE SOUTH       CLEAR LAKE    SD     57226   
4              564 E PIONEER DRIVE          HEPPNER    OR     97836   
..                             ...              ...   ...       ...   
112             704 HOSPITAL DRIVE  CARRIZO SPRINGS    TX     78834   
113           135 EAST SWAN STREET      CENTERVILLE    TN     37033   
114           2600 LOCKWOOD STREET           TAHOKA    TX     79373   
115            55 FOUNDATION DRIVE     FLEMINGSBURG    KY     41041   
116  12950 EAST FREEWAY, SUITE 100          HOUSTON    TX     77015   

        County Name  Phone Number              Hospital Type  \
0             TOOLE  406 434-3200  Critical Access Hospitals   
1    SAN BERNARDINO  909 866-6501  Critical Access Hospitals   
2          CHEYENNE  719 767-5661  Critical Access Hospitals   
3             DEUEL  605 874-2141  Critical Access Hospitals   
4            MORROW  541 676-9133  Critical Access Hospitals   
..              ...           ...                        ...   
112          DIMMIT  830 876-2424  Critical Access Hospitals   
113         HICKMAN  931 729-4271  Critical Access Hospitals   
114            LYNN  806 998-4533  Critical Access Hospitals   
115         FLEMING  606 849-2351  Critical Access Hospitals   
116          HARRIS  713 330-3897       Acute Care Hospitals   

                              Hospital Ownership Facility ID  \
0                             Government - Local      271328   
1    Government - Hospital District or Authority       51335   
2    Government - Hospital District or Authority       61343   
3                 Voluntary non-profit - Private      431307   
4    Government - Hospital District or Authority      381310   
..                                           ...         ...   
112                 Voluntary non-profit - Other      451390   
113               Voluntary non-profit - Private      441300   
114  Government - Hospital District or Authority      451351   
115                                  Proprietary      181332   
116                                  Proprietary      670102   

     Mortality national comparison footnote  \
0                                      16.0   
1                                      16.0   
2                                      16.0   
3                                      16.0   
4                                      16.0   
..                                      ...   
112                                    16.0   
113                                    16.0   
114                                    16.0   
115                                     5.0   
116                                    16.0   

     Safety of care national comparison footnote  \
0                                           16.0   
1                               

#### add the drop table 

In [46]:

# @hidden_cell
# The following code contains the credentials for a file in your IBM Cloud Object Storage.
# You might want to remove those credentials before you share your notebook.
credentials_2 = {
    'IAM_SERVICE_ID': 'iam-ServiceId-b21f1762-43ac-4c08-b713-87cc313601ba',
    'IBM_API_KEY_ID': 'RxZH6HW3U6L9FJUUv4Z53BJeVVYdZ4yvYvPrX7tpEG0P',
    'ENDPOINT': 'https://s3-api.us-geo.objectstorage.service.networklayer.com',
    'IBM_AUTH_ENDPOINT': 'https://iam.cloud.ibm.com/oidc/token',
    'BUCKET': 'hospitalselection-donotdelete-pr-qrw5w2z8u3gbmq',
    'FILE': 'drop table.csv'
}


In [47]:

body = client_8019663306054913a106f7ca1a38c7b3.get_object(Bucket='hospitalselection-donotdelete-pr-qrw5w2z8u3gbmq',Key='drop table.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

df_data_3 = pd.read_csv(body)
df_data_3.head()


Facility Name                       Address  \
0                 CONWAY BEHAVIORAL HEALTH             2255 STURGIS ROAD   
1               SAN JOSE BEHAVIORAL HEALTH  455 SILICON VALLEY BOULEVARD   
2         KAUAI VETERANS MEMORIAL HOSPITAL      4643 WAIMEA CANYON DRIVE   
3  HARDIN COUNTY GENERAL HOSPITAL & CLINIC                  FERRELL ROAD   
4                         PARK CENTER, INC             1909 CAREW STREET   

         City  County Name  Phone Number              Hospital Ownership  \
0      CONWAY     FAULKNER  501 205-0011                     Proprietary   
1    SAN JOSE  SANTA CLARA  669 234-5959                     Proprietary   
2      WAIMEA        KAUAI  808 338-9431              Government - State   
3   ROSICLARE       HARDIN  618 285-6634  Voluntary non-profit - Private   
4  FORT WAYNE        ALLEN  260 481-2700  Voluntary non-profit - Private   

                         Location  
0                             NaN  
1   -121.77291599999998 37.246742  
2   -159.67005500000002 21.960079  
3                             NaN  
4            -85.108024 41.093791

In [48]:
df_data_3.dropna(subset=["Location"], axis=0, inplace=True)

#### merge with original one 

In [49]:
df=pd.merge(df_data_3,df_data_2,on="Location")

In [50]:
df

Facility Name_x                      Address_x  \
0                MARIAS MEDICAL CENTER                   640 PARK AVE   
1       BEAR VALLEY COMMUNITY HOSPITAL               41870 GARSTIN DR   
2              KEEFE MEMORIAL HOSPITAL                 602 N 6TH W ST   
3    SANFORD CLEAR LAKE MEDICAL CENTER         701 THIRD AVENUE SOUTH   
4    PIONEER MEMORIAL HOSPITAL HEPPNER            564 E PIONEER DRIVE   
..                                 ...                            ...   
113           DIMMIT REGIONAL HOSPITAL             704 HOSPITAL DRIVE   
114      SAINT THOMAS HICKMAN HOSPITAL           135 EAST SWAN STREET   
115      LYNN COUNTY HOSPITAL DISTRICT           2600 LOCKWOOD STREET   
116            FLEMING COUNTY HOSPITAL            55 FOUNDATION DRIVE   
117              AD HOSPITAL EAST, LLC  12950 EAST FREEWAY, SUITE 100   

              City_x   County Name_x Phone Number_x  \
0             SHELBY           TOOLE   406 434-3200   
1      BIG BEAR LAKE  SAN BERNARDINO   909 866-6501   
2     CHEYENNE WELLS        CHEYENNE   719 767-5661   
3         CLEAR LAKE           DEUEL   605 874-2141   
4            HEPPNER          MORROW   541 676-9133   
..               ...             ...            ...   
113  CARRIZO SPRINGS          DIMMIT   830 876-2424   
114      CENTERVILLE         HICKMAN   931 729-4271   
115           TAHOKA            LYNN   806 998-4533   
116     FLEMINGSBURG         FLEMING   606 849-2351   
117          HOUSTON          HARRIS   713 330-3897   

                            Hospital Ownership_x  \
0                             Government - Local   
1    Government - Hospital District or Authority   
2    Government - Hospital District or Authority   
3                 Voluntary non-profit - Private   
4    Government - Hospital District or Authority   
..                                           ...   
113                 Voluntary non-profit - Other   
114               Voluntary non-profit - Private   
115  Government - Hospital District or Authority   
116                                  Proprietary   
117                                  Proprietary   

                           Location  Hospital overall rating footnote  \
0             -111.850493 48.509604                              16.0   
1     -116.88834300000002 34.246023                              16.0   
2             -102.355583 38.825887                              16.0   
3               -96.68256 44.752678                              16.0   
4     -119.54763000000001 45.356268                              16.0   
..                              ...                               ...   
113    -99.86209800000002 28.519974                              16.0   
114            -87.460242 35.779539                              16.0   
115           -101.815507 33.166539                              16.0   
116            -83.753094 38.421539                               5.0   
117            -95.200782 29.771136                              16.0   

                       Facility Name_y                      Address_y  ...  \
0                MARIAS MEDICAL CENTER                   640 PARK AVE  ...   
1       BEAR VALLEY COMMUNITY HOSPITAL               41870 GARSTIN DR  ...   
2              KEEFE MEMORIAL HOSPITAL                 602 N 6TH W ST  ...   
3    SANFORD CLEAR LAKE MEDICAL CENTER         701 THIRD AVENUE SOUTH  ...   
4    PIONEER MEMORIAL HOSPITAL HEPPNER            564 E PIONEER DRIVE  ...   
..                                 ...                            ...  ...   
113           DIMMIT REGIONAL HOSPITAL             704 HOSPITAL DRIVE  ...   
114      SAINT THOMAS HICKMAN HOSPITAL           135 EAST SWAN STREET  ...   
115      LYNN COUNTY HOSPITAL DISTRICT           2600 LOCKWOOD STREET  ...   
116            FLEMING COUNTY HOSPITAL            55 FOUNDATION DRIVE  ...   
117              AD HOSPITAL EAST, LLC  12950 EAST FREEWAY, SUITE 100  ...   

                            Hospital Ownership

In [51]:
df.drop('Facility Name_y',axis=1,inplace=True)
df.drop('Address_y',axis=1,inplace=True)
df.drop('City_y',axis=1,inplace=True)
df.drop('County Name_y',axis=1,inplace=True)
df.drop('Phone Number_y',axis=1,inplace=True)
df.drop('Hospital Ownership_y',axis=1,inplace=True)

In [52]:
df

Facility Name_x                      Address_x  \
0                MARIAS MEDICAL CENTER                   640 PARK AVE   
1       BEAR VALLEY COMMUNITY HOSPITAL               41870 GARSTIN DR   
2              KEEFE MEMORIAL HOSPITAL                 602 N 6TH W ST   
3    SANFORD CLEAR LAKE MEDICAL CENTER         701 THIRD AVENUE SOUTH   
4    PIONEER MEMORIAL HOSPITAL HEPPNER            564 E PIONEER DRIVE   
..                                 ...                            ...   
113           DIMMIT REGIONAL HOSPITAL             704 HOSPITAL DRIVE   
114      SAINT THOMAS HICKMAN HOSPITAL           135 EAST SWAN STREET   
115      LYNN COUNTY HOSPITAL DISTRICT           2600 LOCKWOOD STREET   
116            FLEMING COUNTY HOSPITAL            55 FOUNDATION DRIVE   
117              AD HOSPITAL EAST, LLC  12950 EAST FREEWAY, SUITE 100   

              City_x   County Name_x Phone Number_x  \
0             SHELBY           TOOLE   406 434-3200   
1      BIG BEAR LAKE  SAN BERNARDINO   909 866-6501   
2     CHEYENNE WELLS        CHEYENNE   719 767-5661   
3         CLEAR LAKE           DEUEL   605 874-2141   
4            HEPPNER          MORROW   541 676-9133   
..               ...             ...            ...   
113  CARRIZO SPRINGS          DIMMIT   830 876-2424   
114      CENTERVILLE         HICKMAN   931 729-4271   
115           TAHOKA            LYNN   806 998-4533   
116     FLEMINGSBURG         FLEMING   606 849-2351   
117          HOUSTON          HARRIS   713 330-3897   

                            Hospital Ownership_x  \
0                             Government - Local   
1    Government - Hospital District or Authority   
2    Government - Hospital District or Authority   
3                 Voluntary non-profit - Private   
4    Government - Hospital District or Authority   
..                                           ...   
113                 Voluntary non-profit - Other   
114               Voluntary non-profit - Private   
115  Government - Hospital District or Authority   
116                                  Proprietary   
117                                  Proprietary   

                           Location  Hospital overall rating footnote State  \
0             -111.850493 48.509604                              16.0    MT   
1     -116.88834300000002 34.246023                              16.0    CA   
2             -102.355583 38.825887                              16.0    CO   
3               -96.68256 44.752678                              16.0    SD   
4     -119.54763000000001 45.356268                              16.0    OR   
..                              ...                               ...   ...   
113    -99.86209800000002 28.519974                              16.0    TX   
114            -87.460242 35.779539                              16.0    TN   
115           -101.815507 33.166539                              16.0    TX   
116            -83.753094 38.421539                               5.0    KY   
117            -95.200782 29.771136                              16.0    TX   

     ZIP Code              Hospital Type Facility ID  \
0       59474  Critical Access Hospitals      271328   
1       92315  Critical Access Hospitals       51335   
2       80810  Critical Access Hospitals       61343   
3       57226  Critical Access Hospitals      431307   
4       97836  Critical Access Hospitals      381310   
..        ...                        ...         ...   
113     78834  Critical Access Hospitals      451390   
114     37033  Critical Access Hospitals      441300   
115     79373  Critical Access Hospitals      451351   
116     41041  Critical Access Hospitals      181332   
117     77015       Acute Care Hospitals      670102   

     Mortality national comparison footnote  \
0                                      16.0   
1                                      16.0   
2                                      16.0   
3                                      16.0   
4  

#### Now add location data 

In [61]:

# @hidden_cell
# The following code contains the credentials for a file in your IBM Cloud Object Storage.
# You might want to remove those credentials before you share your notebook.
credentials_3 = {
    'IAM_SERVICE_ID': 'iam-ServiceId-b21f1762-43ac-4c08-b713-87cc313601ba',
    'IBM_API_KEY_ID': 'RxZH6HW3U6L9FJUUv4Z53BJeVVYdZ4yvYvPrX7tpEG0P',
    'ENDPOINT': 'https://s3-api.us-geo.objectstorage.service.networklayer.com',
    'IBM_AUTH_ENDPOINT': 'https://iam.cloud.ibm.com/oidc/token',
    'BUCKET': 'hospitalselection-donotdelete-pr-qrw5w2z8u3gbmq',
    'FILE': 'location data2.csv'
}


In [62]:

body = client_8019663306054913a106f7ca1a38c7b3.get_object(Bucket='hospitalselection-donotdelete-pr-qrw5w2z8u3gbmq',Key='location data2.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )

df_data_4 = pd.read_csv(body)
df_data_4.head()


Location   Latitude   Longitude
0                             NaN         NaN        NaN
1   -121.77291599999998 37.246742 -121.772916  37.246742
2   -159.67005500000002 21.960079 -159.670055  21.960079
3                             NaN         NaN        NaN
4            -85.108024 41.093791  -85.108024  41.093791

In [63]:
df_data_4.dropna(subset=["Location"], axis=0, inplace=True)
df_data_4

Location   Latitude   Longitude
1      -121.77291599999998 37.246742 -121.772916  37.246742
2      -159.67005500000002 21.960079 -159.670055  21.960079
4               -85.108024 41.093791  -85.108024  41.093791
8               -94.101386 45.254576  -94.101386  45.254576
9               -71.154678 42.869242  -71.154678  42.869242
...                              ...         ...        ...
5309            -83.227413 37.279027  -83.227413  37.279027
5310            -84.293721 39.336786  -84.293721  39.336786
5311              -95.22854 30.04461  -95.228540  30.044610
5312            -80.254161 26.268779  -80.254161  26.268779
5313   -94.807751 39.776936000000006  -94.807751  39.776936

[4910 rows x 3 columns]

In [64]:
df_data_4.reset_index(drop=True, inplace=True)
df_data_4

Location   Latitude   Longitude
0      -121.77291599999998 37.246742 -121.772916  37.246742
1      -159.67005500000002 21.960079 -159.670055  21.960079
2               -85.108024 41.093791  -85.108024  41.093791
3               -94.101386 45.254576  -94.101386  45.254576
4               -71.154678 42.869242  -71.154678  42.869242
...                              ...         ...        ...
4905            -83.227413 37.279027  -83.227413  37.279027
4906            -84.293721 39.336786  -84.293721  39.336786
4907              -95.22854 30.04461  -95.228540  30.044610
4908            -80.254161 26.268779  -80.254161  26.268779
4909   -94.807751 39.776936000000006  -94.807751  39.776936

[4910 rows x 3 columns]

In [65]:
with_lat_long=df=pd.merge(df,df_data_4,on="Location")
with_lat_long

Facility Name_x                      Address_x  \
0                MARIAS MEDICAL CENTER                   640 PARK AVE   
1       BEAR VALLEY COMMUNITY HOSPITAL               41870 GARSTIN DR   
2              KEEFE MEMORIAL HOSPITAL                 602 N 6TH W ST   
3    SANFORD CLEAR LAKE MEDICAL CENTER         701 THIRD AVENUE SOUTH   
4    PIONEER MEMORIAL HOSPITAL HEPPNER            564 E PIONEER DRIVE   
..                                 ...                            ...   
119           DIMMIT REGIONAL HOSPITAL             704 HOSPITAL DRIVE   
120      SAINT THOMAS HICKMAN HOSPITAL           135 EAST SWAN STREET   
121      LYNN COUNTY HOSPITAL DISTRICT           2600 LOCKWOOD STREET   
122            FLEMING COUNTY HOSPITAL            55 FOUNDATION DRIVE   
123              AD HOSPITAL EAST, LLC  12950 EAST FREEWAY, SUITE 100   

              City_x   County Name_x Phone Number_x  \
0             SHELBY           TOOLE   406 434-3200   
1      BIG BEAR LAKE  SAN BERNARDINO   909 866-6501   
2     CHEYENNE WELLS        CHEYENNE   719 767-5661   
3         CLEAR LAKE           DEUEL   605 874-2141   
4            HEPPNER          MORROW   541 676-9133   
..               ...             ...            ...   
119  CARRIZO SPRINGS          DIMMIT   830 876-2424   
120      CENTERVILLE         HICKMAN   931 729-4271   
121           TAHOKA            LYNN   806 998-4533   
122     FLEMINGSBURG         FLEMING   606 849-2351   
123          HOUSTON          HARRIS   713 330-3897   

                            Hospital Ownership_x  \
0                             Government - Local   
1    Government - Hospital District or Authority   
2    Government - Hospital District or Authority   
3                 Voluntary non-profit - Private   
4    Government - Hospital District or Authority   
..                                           ...   
119                 Voluntary non-profit - Other   
120               Voluntary non-profit - Private   
121  Government - Hospital District or Authority   
122                                  Proprietary   
123                                  Proprietary   

                           Location  Hospital overall rating footnote State  \
0             -111.850493 48.509604                              16.0    MT   
1     -116.88834300000002 34.246023                              16.0    CA   
2             -102.355583 38.825887                              16.0    CO   
3               -96.68256 44.752678                              16.0    SD   
4     -119.54763000000001 45.356268                              16.0    OR   
..                              ...                               ...   ...   
119    -99.86209800000002 28.519974                              16.0    TX   
120            -87.460242 35.779539                              16.0    TN   
121           -101.815507 33.166539                              16.0    TX   
122            -83.753094 38.421539                               5.0    KY   
123            -95.200782 29.771136                              16.0    TX   

     ZIP Code  ... Readmission national comparison footnote  \
0       59474  ...                                      5.0   
1       92315  ...                                      5.0   
2       80810  ...                                     16.0   
3       57226  ...                                      5.0   
4       97836  ...                                      5.0   
..        ...  ...                                      ...   
119     78834  ...                                     16.0   
120     37033  ...                                     16.0   
121     79373  ...                                      5.0   
122     41041  ...                                      5.0   
123     77015  ...                                     16.0   

    Patient experience national comparison footnote  \
0                                              16.0   
1                                           

In [66]:
with_lat_long[["Latitude _x"]]

Latitude _x
0    -111.850493
1    -116.888343
2    -102.355583
3     -96.682560
4    -119.547630
..           ...
119   -99.862098
120   -87.460242
121  -101.815507
122   -83.753094
123   -95.200782

[124 rows x 1 columns]

In [67]:
with_lat_long.drop('Latitude _y',axis=1,inplace=True)
with_lat_long.drop('Longitude_y',axis=1,inplace=True)

In [68]:
with_lat_long

Facility Name_x                      Address_x  \
0                MARIAS MEDICAL CENTER                   640 PARK AVE   
1       BEAR VALLEY COMMUNITY HOSPITAL               41870 GARSTIN DR   
2              KEEFE MEMORIAL HOSPITAL                 602 N 6TH W ST   
3    SANFORD CLEAR LAKE MEDICAL CENTER         701 THIRD AVENUE SOUTH   
4    PIONEER MEMORIAL HOSPITAL HEPPNER            564 E PIONEER DRIVE   
..                                 ...                            ...   
119           DIMMIT REGIONAL HOSPITAL             704 HOSPITAL DRIVE   
120      SAINT THOMAS HICKMAN HOSPITAL           135 EAST SWAN STREET   
121      LYNN COUNTY HOSPITAL DISTRICT           2600 LOCKWOOD STREET   
122            FLEMING COUNTY HOSPITAL            55 FOUNDATION DRIVE   
123              AD HOSPITAL EAST, LLC  12950 EAST FREEWAY, SUITE 100   

              City_x   County Name_x Phone Number_x  \
0             SHELBY           TOOLE   406 434-3200   
1      BIG BEAR LAKE  SAN BERNARDINO   909 866-6501   
2     CHEYENNE WELLS        CHEYENNE   719 767-5661   
3         CLEAR LAKE           DEUEL   605 874-2141   
4            HEPPNER          MORROW   541 676-9133   
..               ...             ...            ...   
119  CARRIZO SPRINGS          DIMMIT   830 876-2424   
120      CENTERVILLE         HICKMAN   931 729-4271   
121           TAHOKA            LYNN   806 998-4533   
122     FLEMINGSBURG         FLEMING   606 849-2351   
123          HOUSTON          HARRIS   713 330-3897   

                            Hospital Ownership_x  \
0                             Government - Local   
1    Government - Hospital District or Authority   
2    Government - Hospital District or Authority   
3                 Voluntary non-profit - Private   
4    Government - Hospital District or Authority   
..                                           ...   
119                 Voluntary non-profit - Other   
120               Voluntary non-profit - Private   
121  Government - Hospital District or Authority   
122                                  Proprietary   
123                                  Proprietary   

                           Location  Hospital overall rating footnote State  \
0             -111.850493 48.509604                              16.0    MT   
1     -116.88834300000002 34.246023                              16.0    CA   
2             -102.355583 38.825887                              16.0    CO   
3               -96.68256 44.752678                              16.0    SD   
4     -119.54763000000001 45.356268                              16.0    OR   
..                              ...                               ...   ...   
119    -99.86209800000002 28.519974                              16.0    TX   
120            -87.460242 35.779539                              16.0    TN   
121           -101.815507 33.166539                              16.0    TX   
122            -83.753094 38.421539                               5.0    KY   
123            -95.200782 29.771136                              16.0    TX   

     ZIP Code  ... Mortality national comparison footnote  \
0       59474  ...                                   16.0   
1       92315  ...                                   16.0   
2       80810  ...                                   16.0   
3       57226  ...                                   16.0   
4       97836  ...                                   16.0   
..        ...  ...                                    ...   
119     78834  ...                                   16.0   
120     37033  ...                                   16.0   
121     79373  ...                                   16.0   
122     41041  ...                                    5.0   
123     77015  ...                                   16.0   

    Safety of care national comparison footnote  \
0                                          16.0   
1                                          16.0   
2                        

In [69]:
with_lat_long.rename(columns={'Latitude _x':'Latitude','Longitude_x':'Longitude'},inplace=True)
with_lat_long

Facility Name_x                      Address_x  \
0                MARIAS MEDICAL CENTER                   640 PARK AVE   
1       BEAR VALLEY COMMUNITY HOSPITAL               41870 GARSTIN DR   
2              KEEFE MEMORIAL HOSPITAL                 602 N 6TH W ST   
3    SANFORD CLEAR LAKE MEDICAL CENTER         701 THIRD AVENUE SOUTH   
4    PIONEER MEMORIAL HOSPITAL HEPPNER            564 E PIONEER DRIVE   
..                                 ...                            ...   
119           DIMMIT REGIONAL HOSPITAL             704 HOSPITAL DRIVE   
120      SAINT THOMAS HICKMAN HOSPITAL           135 EAST SWAN STREET   
121      LYNN COUNTY HOSPITAL DISTRICT           2600 LOCKWOOD STREET   
122            FLEMING COUNTY HOSPITAL            55 FOUNDATION DRIVE   
123              AD HOSPITAL EAST, LLC  12950 EAST FREEWAY, SUITE 100   

              City_x   County Name_x Phone Number_x  \
0             SHELBY           TOOLE   406 434-3200   
1      BIG BEAR LAKE  SAN BERNARDINO   909 866-6501   
2     CHEYENNE WELLS        CHEYENNE   719 767-5661   
3         CLEAR LAKE           DEUEL   605 874-2141   
4            HEPPNER          MORROW   541 676-9133   
..               ...             ...            ...   
119  CARRIZO SPRINGS          DIMMIT   830 876-2424   
120      CENTERVILLE         HICKMAN   931 729-4271   
121           TAHOKA            LYNN   806 998-4533   
122     FLEMINGSBURG         FLEMING   606 849-2351   
123          HOUSTON          HARRIS   713 330-3897   

                            Hospital Ownership_x  \
0                             Government - Local   
1    Government - Hospital District or Authority   
2    Government - Hospital District or Authority   
3                 Voluntary non-profit - Private   
4    Government - Hospital District or Authority   
..                                           ...   
119                 Voluntary non-profit - Other   
120               Voluntary non-profit - Private   
121  Government - Hospital District or Authority   
122                                  Proprietary   
123                                  Proprietary   

                           Location  Hospital overall rating footnote State  \
0             -111.850493 48.509604                              16.0    MT   
1     -116.88834300000002 34.246023                              16.0    CA   
2             -102.355583 38.825887                              16.0    CO   
3               -96.68256 44.752678                              16.0    SD   
4     -119.54763000000001 45.356268                              16.0    OR   
..                              ...                               ...   ...   
119    -99.86209800000002 28.519974                              16.0    TX   
120            -87.460242 35.779539                              16.0    TN   
121           -101.815507 33.166539                              16.0    TX   
122            -83.753094 38.421539                               5.0    KY   
123            -95.200782 29.771136                              16.0    TX   

     ZIP Code  ... Mortality national comparison footnote  \
0       59474  ...                                   16.0   
1       92315  ...                                   16.0   
2       80810  ...                                   16.0   
3       57226  ...                                   16.0   
4       97836  ...                                   16.0   
..        ...  ...                                    ...   
119     78834  ...                                   16.0   
120     37033  ...                                   16.0   
121     79373  ...                                   16.0   
122     41041  ...                                    5.0   
123     77015  ...                                   16.0   

    Safety of care national comparison footnote  \
0                                          16.0   
1                                          16.0   
2                        

In [70]:
with_lat_long.rename(columns={'County Name_x':'County_Name','Hospital Type':'Hospital_Type'},inplace=True)
with_lat_long

Facility Name_x                      Address_x  \
0                MARIAS MEDICAL CENTER                   640 PARK AVE   
1       BEAR VALLEY COMMUNITY HOSPITAL               41870 GARSTIN DR   
2              KEEFE MEMORIAL HOSPITAL                 602 N 6TH W ST   
3    SANFORD CLEAR LAKE MEDICAL CENTER         701 THIRD AVENUE SOUTH   
4    PIONEER MEMORIAL HOSPITAL HEPPNER            564 E PIONEER DRIVE   
..                                 ...                            ...   
119           DIMMIT REGIONAL HOSPITAL             704 HOSPITAL DRIVE   
120      SAINT THOMAS HICKMAN HOSPITAL           135 EAST SWAN STREET   
121      LYNN COUNTY HOSPITAL DISTRICT           2600 LOCKWOOD STREET   
122            FLEMING COUNTY HOSPITAL            55 FOUNDATION DRIVE   
123              AD HOSPITAL EAST, LLC  12950 EAST FREEWAY, SUITE 100   

              City_x     County_Name Phone Number_x  \
0             SHELBY           TOOLE   406 434-3200   
1      BIG BEAR LAKE  SAN BERNARDINO   909 866-6501   
2     CHEYENNE WELLS        CHEYENNE   719 767-5661   
3         CLEAR LAKE           DEUEL   605 874-2141   
4            HEPPNER          MORROW   541 676-9133   
..               ...             ...            ...   
119  CARRIZO SPRINGS          DIMMIT   830 876-2424   
120      CENTERVILLE         HICKMAN   931 729-4271   
121           TAHOKA            LYNN   806 998-4533   
122     FLEMINGSBURG         FLEMING   606 849-2351   
123          HOUSTON          HARRIS   713 330-3897   

                            Hospital Ownership_x  \
0                             Government - Local   
1    Government - Hospital District or Authority   
2    Government - Hospital District or Authority   
3                 Voluntary non-profit - Private   
4    Government - Hospital District or Authority   
..                                           ...   
119                 Voluntary non-profit - Other   
120               Voluntary non-profit - Private   
121  Government - Hospital District or Authority   
122                                  Proprietary   
123                                  Proprietary   

                           Location  Hospital overall rating footnote State  \
0             -111.850493 48.509604                              16.0    MT   
1     -116.88834300000002 34.246023                              16.0    CA   
2             -102.355583 38.825887                              16.0    CO   
3               -96.68256 44.752678                              16.0    SD   
4     -119.54763000000001 45.356268                              16.0    OR   
..                              ...                               ...   ...   
119    -99.86209800000002 28.519974                              16.0    TX   
120            -87.460242 35.779539                              16.0    TN   
121           -101.815507 33.166539                              16.0    TX   
122            -83.753094 38.421539                               5.0    KY   
123            -95.200782 29.771136                              16.0    TX   

     ZIP Code  ... Mortality national comparison footnote  \
0       59474  ...                                   16.0   
1       92315  ...                                   16.0   
2       80810  ...                                   16.0   
3       57226  ...                                   16.0   
4       97836  ...                                   16.0   
..        ...  ...                                    ...   
119     78834  ...                                   16.0   
120     37033  ...                                   16.0   
121     79373  ...                                   16.0   
122     41041  ...                                    5.0   
123     77015  ...                                   16.0   

    Safety of care national comparison footnote  \
0                                          16.0   
1                                          16.0   
2                        

instal some libary such as BeautifulSoup4 and geopy

In [72]:
!pip install BeautifulSoup4
!pip install geopy

#### imort some libary for visualization purpose

In [73]:
import requests
import folium 
import matplotlib.cm as cm
import matplotlib.colors as colors
import json

from urllib.request import urlopen
from geopy.geocoders import Nominatim 
from pandas.io.json import json_normalize
from sklearn.cluster import KMeans
from bs4 import BeautifulSoup

In [74]:
user_wish = with_lat_long[with_lat_long['Hospital_Type'].str.contains('Critical Access Hospitals')].reset_index(drop=True)
user_wish.head(116)

Facility Name_x                 Address_x  \
0               MARIAS MEDICAL CENTER              640 PARK AVE   
1      BEAR VALLEY COMMUNITY HOSPITAL          41870 GARSTIN DR   
2             KEEFE MEMORIAL HOSPITAL            602 N 6TH W ST   
3   SANFORD CLEAR LAKE MEDICAL CENTER    701 THIRD AVENUE SOUTH   
4   PIONEER MEMORIAL HOSPITAL HEPPNER       564 E PIONEER DRIVE   
..                                ...                       ...   
92     AVERA FLANDREAU HOSPITAL - CAH  214 NORTH PRAIRIE STREET   
93           DIMMIT REGIONAL HOSPITAL        704 HOSPITAL DRIVE   
94      SAINT THOMAS HICKMAN HOSPITAL      135 EAST SWAN STREET   
95      LYNN COUNTY HOSPITAL DISTRICT      2600 LOCKWOOD STREET   
96            FLEMING COUNTY HOSPITAL       55 FOUNDATION DRIVE   

             City_x     County_Name Phone Number_x  \
0            SHELBY           TOOLE   406 434-3200   
1     BIG BEAR LAKE  SAN BERNARDINO   909 866-6501   
2    CHEYENNE WELLS        CHEYENNE   719 767-5661   
3        CLEAR LAKE           DEUEL   605 874-2141   
4           HEPPNER          MORROW   541 676-9133   
..              ...             ...            ...   
92        FLANDREAU           MOODY   605 997-2433   
93  CARRIZO SPRINGS          DIMMIT   830 876-2424   
94      CENTERVILLE         HICKMAN   931 729-4271   
95           TAHOKA            LYNN   806 998-4533   
96     FLEMINGSBURG         FLEMING   606 849-2351   

                           Hospital Ownership_x  \
0                            Government - Local   
1   Government - Hospital District or Authority   
2   Government - Hospital District or Authority   
3                Voluntary non-profit - Private   
4   Government - Hospital District or Authority   
..                                          ...   
92               Voluntary non-profit - Private   
93                 Voluntary non-profit - Other   
94               Voluntary non-profit - Private   
95  Government - Hospital District or Authority   
96                                  Proprietary   

                          Location  Hospital overall rating footnote State  \
0            -111.850493 48.509604                              16.0    MT   
1    -116.88834300000002 34.246023                              16.0    CA   
2            -102.355583 38.825887                              16.0    CO   
3              -96.68256 44.752678                              16.0    SD   
4    -119.54763000000001 45.356268                              16.0    OR   
..                             ...                               ...   ...   
92   -96.590989 44.050568000000005                              16.0    SD   
93    -99.86209800000002 28.519974                              16.0    TX   
94            -87.460242 35.779539                              16.0    TN   
95           -101.815507 33.166539                              16.0    TX   
96            -83.753094 38.421539                               5.0    KY   

    ZIP Code  ... Mortality national comparison footnote  \
0      59474  ...                                   16.0   
1      92315  ...                                   16.0   
2      80810  ...                                   16.0   
3      57226  ...                                   16.0   
4      97836  ...                                   16.0   
..       ...  ...                                    ...   
92     57028  ...                                   16.0   
93     78834  ...                                   16.0   
94     37033  ...                                   16.0   
95     79373  ...                                   16.0   
96     41041  ...                                    5.0   

   Safety of care national comparison footnote  \
0                                         16.0   
1                                         16.0   
2                                         16.0   
3                                         16.0   
4                                         16.0 

#### enter your four square credential 

In [92]:
CLIENT_ID = 'ZUWOAQTQEIPGE0YB0RCGUZFEFT5QR13XGJWR0IA3RDWLEU14' # your Foursquare ID
CLIENT_SECRET = 'EB44FEKX5WFX05JEIWJKHG2ZFSMUKLOOECTPNRVYJDTXUCMS' # your Foursquare Secret
ACCESS_TOKEN = 'ZXF0CRQEV0YSULND1TMZVE0YY3LVNVA5A1T5XWBJBICDAA4B' # your FourSquare Access Token
VERSION = '20180604'
LIMIT = 30
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: ZUWOAQTQEIPGE0YB0RCGUZFEFT5QR13XGJWR0IA3RDWLEU14
CLIENT_SECRET:EB44FEKX5WFX05JEIWJKHG2ZFSMUKLOOECTPNRVYJDTXUCMS


now explore the first area where hospital type Critical Access Hospitals to explore the first are useuser_wish.loc[0,'latitude']

In [156]:
latitude = user_wish.loc[0, 'Latitude'] 
longitude = user_wish.loc[0, 'Longitude'] 
print(latitude, longitude)

40.7149555 -74.0153365


In [160]:
search_query = 'hospital search okay'
radius = 500
print(search_query + ' .... OK!')

hospital search okay .... OK!


In [163]:
url = 'https://api.foursquare.com/v2/venues/search?client_id=ZUWOAQTQEIPGE0YB0RCGUZFEFT5QR13XGJWR0IA3RDWLEU14&client_secret=EB44FEKX5WFX05JEIWJKHG2ZFSMUKLOOECTPNRVYJDTXUCMS&ll=40.7149555,-74.0153365&oauth_token=ZXF0CRQEV0YSULND1TMZVE0YY3LVNVA5A1T5XWBJBICDAA4B&v=20180604&query=Italian&radius=500&limit=30'.format(CLIENT_ID, CLIENT_SECRET, latitude, longitude,ACCESS_TOKEN, VERSION, search_query, radius, LIMIT)
url

'https://api.foursquare.com/v2/venues/search?client_id=ZUWOAQTQEIPGE0YB0RCGUZFEFT5QR13XGJWR0IA3RDWLEU14&client_secret=EB44FEKX5WFX05JEIWJKHG2ZFSMUKLOOECTPNRVYJDTXUCMS&ll=40.7149555,-74.0153365&oauth_token=ZXF0CRQEV0YSULND1TMZVE0YY3LVNVA5A1T5XWBJBICDAA4B&v=20180604&query=Italian&radius=500&limit=30'

In [171]:
results = requests.get(url).json()


In [165]:
# assign relevant part of JSON to venues
venues = results['response']['venues']

# tranform venues into a dataframe
dataframe = json_normalize(venues)
dataframe.head()

/opt/conda/envs/Python-3.7-main/lib/python3.7/site-packages/ipykernel/__main__.py:5: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead


id                             name  \
0  4fa862b3e4b0ebff2f749f06        Harry's Italian Pizza Bar   
1  4f3232e219836c91c7bfde94  Conca Cucina Italian Restaurant   
2  3fd66200f964a520f4e41ee3                             Ecco   

                                          categories    referralId  hasPerk  \
0  [{'id': '4bf58dd8d48988d1ca941735', 'name': 'P...  v-1607985313    False   
1  [{'id': '4d4b7105d754a06374d81259', 'name': 'F...  v-1607985313    False   
2  [{'id': '4bf58dd8d48988d110941735', 'name': 'I...  v-1607985313    False   

  location.address  location.lat  location.lng  \
0    225 Murray St     40.715218    -74.014739   
1    63 W Broadway     40.714484    -74.009806   
2  124 Chambers St     40.715337    -74.008848   

                             location.labeledLatLngs  location.distance  \
0  [{'label': 'display', 'lat': 40.71521779064671...                 58   
1  [{'label': 'display', 'lat': 40.71448400000000...                469   
2  [{'label': 'display', 'lat': 40.71533713859952...                549   

  location.postalCode location.cc location.city location.state  \
0               10282          US      New York             NY   
1               10007          US      New York             NY   
2               10007          US      New York             NY   

  location.country                          location.formattedAddress  \
0    United States  [225 Murray St, New York, NY 10282, United Sta...   
1    United States  [63 W Broadway, New York, NY 10007, United Sta...   
2    United States  [124 Chambers St (btwn Church St & W Broadway)...   

          location.crossStreet  
0                          NaN  
1                          NaN  
2  btwn Church St & W Broadway

In [174]:
# keep only columns that include venue name, and anything that is associated with location
filtered_columns = ['name', 'categories'] + [col for col in dataframe.columns if col.startswith('location.')] + ['id']
dataframe_filtered = dataframe.loc[:, filtered_columns]

# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

# filter the category for each row
dataframe_filtered['categories'] = dataframe_filtered.apply(get_category_type, axis=1)

# clean column names by keeping only last term
dataframe_filtered.columns = [column.split('.')[-1] for column in dataframe_filtered.columns]



In [187]:
dataframe_filtered

categories        lat        lng  \
0         Pizza Place  40.715218 -74.014739   
1                Food  40.714484 -74.009806   
2  Italian Restaurant  40.715337 -74.008848   

                                      labeledLatLngs  distance postalCode  cc  \
0  [{'label': 'display', 'lat': 40.71521779064671...        58      10282  US   
1  [{'label': 'display', 'lat': 40.71448400000000...       469      10007  US   
2  [{'label': 'display', 'lat': 40.71533713859952...       549      10007  US   

       city state        country                  crossStreet  
0  New York    NY  United States                          NaN  
1  New York    NY  United States                          NaN  
2  New York    NY  United States  btwn Church St & W Broadway

In [181]:
dataframe_filtered.categories

0           Pizza Place
1                  Food
2    Italian Restaurant
Name: categories, dtype: object

In [118]:
venues_map = folium.Map(location=[latitude, longitude], zoom_start=13) # generate map centred around the Conrad Hotel

# add a red circle marker to represent the Conrad Hotel
folium.CircleMarker(
    [latitude, longitude],
    radius=10,
    color='red',
    popup='Conrad Hotel',
    fill = True,
    fill_color = 'red',
    fill_opacity = 0.6
).add_to(venues_map)

# add the Italian restaurants as blue circle markers
for lat, lng, label in zip(dataframe_filtered.lat, dataframe_filtered.lng, dataframe_filtered.categories):
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        color='blue',
        popup=label,
        fill = True,
        fill_color='blue',
        fill_opacity=0.6
    ).add_to(venues_map)

# display map
venues_map